api/provider/aws/aws_test/load_balancer_create_test.go

```diff
@@ -0,0 +1,257 @@
+	}
+
+	readSGHelper(mockAWS, "l0-test-env_id-env", "env_sg")
+	createSGHelper(t, mockAWS, "l0-test-lb_id-lb", "vpc_id")
+	readSGHelper(mockAWS, "l0-test-lb_id-lb", "lb_sg")
+
+	for _, port := range req.Ports {
+		ingressInput := &ec2.AuthorizeSecurityGroupIngressInput{}
+		ingressInput.SetGroupId("lb_sg")
+		ingressInput.SetCidrIp("0.0.0.0/0")
+		ingressInput.SetIpProtocol("TCP")
+		ingressInput.SetFromPort(int64(port.HostPort))
+		ingressInput.SetToPort(int64(port.HostPort))
+
+		mockAWS.EC2.EXPECT().
+			AuthorizeSecurityGroupIngress(ingressInput).
+			Return(&ec2.AuthorizeSecurityGroupIngressOutput{}, nil)
+	}
+
+	iamRoleInput := &iam.CreateRoleInput{}
+	iamRoleInput.SetRoleName("l0-test-lb_id-lb")
+	iamRoleInput.SetAssumeRolePolicyDocument(provider.DEFAULT_ASSUME_ROLE_POLICY)
+
+	mockAWS.IAM.EXPECT().
+		CreateRole(iamRoleInput).
+		Return(&iam.CreateRoleOutput{}, nil)
+
+	renderedLBRolePolicy, err := provider.RenderLoadBalancerRolePolicy(
+		"region",
+		"123456789012",
+		"l0-test-lb_id",
+		provider.DEFAULT_LB_ROLE_POLICY_TEMPLATE)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	putIAMPolicyInput := &iam.PutRolePolicyInput{}
+	putIAMPolicyInput.SetPolicyName("l0-test-lb_id-lb")
+	putIAMPolicyInput.SetRoleName("l0-test-lb_id-lb")
+	putIAMPolicyInput.SetPolicyDocument(renderedLBRolePolicy)
+
+	mockAWS.IAM.EXPECT().
+		PutRolePolicy(putIAMPolicyInput).
+		Return(&iam.PutRolePolicyOutput{}, nil)
+
+	listeners := make([]*elb.Listener, len(req.Ports))
+	for i, port := range req.Ports {
+		listener := &elb.Listener{}
+		listener.SetProtocol(port.Protocol)
+		listener.SetLoadBalancerPort(port.HostPort)
+		listener.SetInstancePort(port.ContainerPort)
+
+		if port.CertificateName != "" {
+			serverCertificateMetadataList := []*iam.ServerCertificateMetadata{
+				&iam.ServerCertificateMetadata{
+					Arn: aws.String(port.CertificateName),
+					ServerCertificateName: aws.String(port.CertificateName),
+				},
+			}
+
+			listServerCertificatesOutput := &iam.ListServerCertificatesOutput{}
+			listServerCertificatesOutput.SetServerCertificateMetadataList(serverCertificateMetadataList)
+
+			mockAWS.IAM.EXPECT().
+				ListServerCertificates(&iam.ListServerCertificatesInput{}).
+				Return(listServerCertificatesOutput, nil)
+
+			listener.SetSSLCertificateId(port.CertificateName)
+		}
+
+		switch strings.ToLower(port.Protocol) {
+		case "http", "https":
+			listener.SetInstanceProtocol("http")
+		case "tcp", "ssl":
+			listener.SetInstanceProtocol("tcp")
+		}
+
+		listeners[i] = listener
+	}
+
+	createLoadBalancerInput := &elb.CreateLoadBalancerInput{}
+	createLoadBalancerInput.SetLoadBalancerName("l0-test-lb_id")
+	createLoadBalancerInput.SetScheme("internet-facing")
+	createLoadBalancerInput.SetSecurityGroups([]*string{aws.String("env_sg"), aws.String("lb_sg")})
+	createLoadBalancerInput.SetSubnets([]*string{aws.String("pub1"), aws.String("pub2")})
+	createLoadBalancerInput.SetListeners(listeners)
+
+	validateFN := func(input *elb.CreateLoadBalancerInput) {
```

Comment:
this is redundant - the `Listeners` field is already checked for equality below at `CreateLoadBalancer(createLoadBalancerInput).`